virttest/utils_test/qemu/migration.py

```diff
@@ -1377,468 +1211,459 @@ def mig_wrapper(vm, cancel_delay, dsthost, vm_ports,
-        mem_sys_path = "/sys/devices/system/memory/%s" % memory
-        mem_state_path = os.path.join(mem_sys_path, 'state')
-        session = self.get_session(vm)
-        session.cmd("echo '%s' > %s" % (operation, mem_state_path))
-        output = session.cmd_output_safe("cat %s" % mem_state_path)
-        if operation not in output:
-            return exceptions.TestFail("Fail to %s %s" % (operation, memory))
-
-    def get_memory_state(self, vm, memory):
-        """Get memorys state in guest OS"""
-        mem_sys_path = "/sys/devices/system/memory/%s" % memory
-        mem_state_path = os.path.join(mem_sys_path, 'state')
-        session = self.get_session(vm)
-        status, output = session.cmd_status_output("cat %s" % mem_state_path)
-        if status != 0:
-            raise exceptions.TestError("Fail to read %s state" % memory)
-        return output.strip()
-
-    def get_offline_memorys(self, vm):
-        """Get unusable memory in guest OS"""
-        def is_offline_memory(x):
-            return self.get_memory_state(vm, x) == 'offline'
-
-        memorys = self.get_all_memorys(vm)
-        return set(filter(is_offline_memory, memorys))
-
-    def get_all_memorys(self, vm):
-        """Get all memorys detected in guest OS"""
-        mem_sys_path = "/sys/devices/system/memory"
-        cmd = "ls %s | grep memory" % mem_sys_path
-        session = self.get_session(vm)
-        output = session.cmd_output_safe(cmd, timeout=90)
-        return set([_ for _ in output.splitlines() if _])
+            o = vm.monitor.info("migrate")
+            if isinstance(o, str):
+                return ("Migration status: cancelled" in o or
+                        "Migration status: canceled" in o)
+            else:
+                return (o.get("status") == "cancelled" or
+                        o.get("status") == "canceled")
+
+    @error.context_aware
+    def clean_up(self, kill_bg_cmd, vm):
+
+        """
+        kill background cmd on dst after migration
+
+        :param kill_bg_cmd: cmd for kill background test
+        :param vm:  vm object
+        """
+
+        error.context("Kill the background test by '%s' in guest"
+                      "." % kill_bg_cmd, logging.info)
+        session = vm.wait_for_login(timeout=self.login_timeout)
+        if session.cmd_status(self.check_running_cmd) != 0:
+            logging.info("The background test in guest is finished, "
+                         "no need to kill.")
+        else:
+            try:
+                s, o = session.cmd_status_output(kill_bg_cmd)
+                logging.info("The output after run kill command: %r" % o)
+                if "No such process" in o or "not found" in o \
+                        or "no running instance" in o:
+                    if session.cmd_status(self.check_running_cmd) != 0:
+                        logging.info("The background test in guest is "
+                                     "finished before kill it.")
```

Comment:
sometimes, the background test is finished when try to kill it. It seldom happened, in order to handle it, check background test again.